src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -46,111 +46,128 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
-            var localDeclarationToLambda = new Dictionary<LocalDeclarationStatementSyntax, LambdaExpressionSyntax>();
+            var nodesFromDiagnostics = new List<(
+                LocalDeclarationStatementSyntax declaration,
+                AnonymousFunctionExpressionSyntax function,
+                List<InvocationExpressionSyntax> invocations)>();
+
             var nodesToTrack = new HashSet<SyntaxNode>();
```

Comment:
speaking of which, why is this a hash set? it's also only used for iterating to add those to nodesToTrack. is there a concern for finding duplications? but if there are duplications then it's probably a bug in the analyzer

Reply:
Simplicity.  You can feel free to use another type if you want.

Reply:
i'll leave it there, but why is `new HashSet<SyntaxNode>` simpler than `new List<SyntaxNode>`?

Reply:
It's simpler (to me) because it means i don't even have to think about what might go wrong if i get a duplicate.  it just can't happen.

Reply:
Other's may think a list is simpler as it may not mask a duplication bug.  To each their own :)